### PR TITLE
fix: list widget item collapse toggle

### LINKED ISSUE #3619: new release breaks list setup I had in yaml
**Describe the bug**
List widget broken on new release.
The CMS was working fine 2 days ago, but I noticed there was a new release. It's loaded through CDN as `<script src="https://unpkg.com/netlify-cms@^2.0.0/dist/netlify-cms.js"></script>`

The List widget that I use (find yaml below) doesn't show correctly anymore in the CMS and when I try to save I get the error:
`ObjectControl.js:65 Uncaught TypeError: this.componentValidate[e.get(...)] is not a function
    at ObjectControl.js:65
    at it.__iterate (immutable.js:2206)
    at it.forEach (immutable.js:4381)
    at ObjectControl.js:63
    at ListControl.js:248
    at Array.forEach (<anonymous>)
    at Object.galleries (ListControl.js:247)
    at EditorControlPane.js:48
    at it.__iterate (immutable.js:2206)
    at it.forEach (immutable.js:4381)`

Everything is fine when I remove that gallery list and also when I include my locally cms.js that uses the previous release.
Probable something has changed in the yaml setup?
I couldn't make out from the multiple changelogs what has changed. How best to prevent such issue in the future, breaking a live site? I guess include a local build :)

**To Reproduce**
I can make a further setup if needed, but I guess it should be clear from my yaml setup and what has changed in the list widget or something related to the yaml definition in the new release.

**Applicable Versions:**

 - Netlify CMS version: 2.10.45
 - Git provider: GitHub
 - OS: Linux
 - Browser version: any

**CMS configuration**
```yaml
collections:
  - name: 'issues' # Used in routes, e.g., /admin/collections/blog
    label: 'Uitgaves' # Used in the UI
    folder: 'content/issues' # The path to the folder where the documents are stored
    format: 'json'
    create: true # Allow users to create new documents in this collection
    slug: '{{year}}-{{month}}-{{day}}-{{slug}}' # Filename template, e.g., YYYY-MM-DD-title.md
    sortableFields: ['commit_date', 'title', 'order']
    fields: # The fields for each document, usually in front matter
      - {label: Order, name: order, widget: number, valueType: int, required: false}
      - {label: Draft, name: draft, widget: boolean, default: false, required: false}
      - {label: Title, name: title, widget: string, required: true}
      - {label: Subtitle, name: subtitle, widget: string, required: false}
      - {label: Alternatieve button text, name: button_text, widget: string, required: false}
      #- {label: Stripe SKU ID, name: sku, widget: string, required: false}
      - {label: Prijs, name: price, widget: number, valueType: float, hint: gebruik punt ipv comma voor centen, required: false}
      - {label: Checkout mededeling, name: checkout_text, widget: string, required: false}
      - {label: Content, name: body, widget: markdown, required: false}
      - {label: Achtergrondkleur, name: block_color, widget: string, hint: HEX code invullen, required: false}
      - {label: Cover, name: image, widget: image, required: false}
      - {label: Galleries, name: galleries, widget: list, optional: true, collapsed: true, fields: [
          {label: Tekstje, name: title, widget: markdown, required: false},
          {label: Draft, name: draft, widget: boolean, default: false, required: false},
          {label: Only wide images?, name: wide, widget: boolean, default: false, required: false},
          {label: Plaatjes, name: gallery, widget: list, optional: true, collapsed: true, fields: [
            {label: Image, name: image, widget: image},
            {label: Caption, name: caption, widget: string, required: false}
          ]}
        ]}
```
The problem is in the last `name: galleries, widget: list,` item.
image

### PR DESCRIPTION
**Pull request by barthc**
Closes #3619

### CODE REVIEW

**Comment by barthc** [comment]
Looking into it.

File: packages/netlify-cms-widget-object/src/ObjectControl.js [hunk 1]
```diff
@@ -141,7 +144,15 @@ export default class ObjectControl extends React.Component {
...
               )}
-              {collapsed ? null : this.renderFields(multiFields, singleField)}
+              <div
+                className={cx({
+                  [css`
+                    ${styleStrings.collapsedObjectControl}
+                  `]: collapsed,
+                })}
+              >
+                {this.renderFields(multiFields, singleField)}
```
Comment (barthc) [inline packages/netlify-cms-widget-object/src/ObjectControl.js:154]: Also a performance fix here, previously any  time the `object` widget toggles, it discards the old form and renders a new one.

Comment (erezrokah) [reply packages/netlify-cms-widget-object/src/ObjectControl.js:154]: Nice fix 👍

File: packages/netlify-cms-widget-list/src/ListControl.js [hunk 2]
```diff
@@ -411,6 +411,7 @@ export default class ListControl extends React.Component {
               ref={this.processControlRef}
               controlRef={controlRef}
               validationKey={key}
+              collapsed={collapsed}
```
Comment (hanneskuettner) [inline packages/netlify-cms-widget-list/src/ListControl.js:414]: Quick question (even after this has been merged).

Isn't the hiding of object controls already handled a few lines above this through the `display: none` of the `collapsedObjectControl` style?

https://github.com/netlify/netlify-cms/blob/a9d0699a82a25d79dd99af4f1ed9643ce7bba220/packages/netlify-cms-widget-list/src/ListControl.js#L396-L400

**Pull request merged**